### PR TITLE
Add bussines logic, start using postgres and create the Flight entity

### PR DESCRIPTION
**Pull request by JofielB**
## Summary
Start working with Postgres, start the creation of the entity Flight and its CRUD and some refactoring.

## Changes
- Add the Postgres dependency
- Set some configurations
- Start using Postgres
- Add some business logic to the create airport endpoint
- Refactor some parts. of the code
- Create the flight entity, repo, service, and controller
- Create a custom exception: FlightNotFound
- Create the ManyToOne bidirectional relationship between Airport and Flight, so we can know which flight belong to an airport

## Target
`main`

### CODE REVIEW

File: src/main/java/com/example/flight/controllers/FlightController.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
+package com.example.flight.controllers;
+
+import com.example.flight.entities.Flight;
+import com.example.flight.services.FlightService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+//Todo fix
```
Comment (janvmusic) [inline src/main/java/com/example/flight/controllers/FlightController.java]: Fix what? Imagine another dev comes to work on this. What is the purpose of this TODO?

File: src/main/java/com/example/flight/controllers/FlightController.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    }
+
+    @GetMapping
+    public List<Flight> getAllFlights() {
+        return service.getAllFlights();
+    }
+
+    @PostMapping
+    public Flight createFlight(@RequestBody Flight flight) {
+        return service.createFlight(flight);
```
Comment (janvmusic) [inline src/main/java/com/example/flight/controllers/FlightController.java]: Not required, this should be coming through a 3rd party service and users should not be able to create flights

File: src/main/java/com/example/flight/controllers/FlightController.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    }
+
+    @GetMapping("{id}")
+    public Flight getFlight(@PathVariable Long id) {
```
Comment (janvmusic) [inline src/main/java/com/example/flight/controllers/FlightController.java]: MInor: Maybe `getFlightDetails`?

File: src/main/java/com/example/flight/controllers/FlightController.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        return service.getFlightById(id);
+    }
+
+//    @PutMapping("{id}")
```
Comment (janvmusic) [inline src/main/java/com/example/flight/controllers/FlightController.java]: remove

File: src/main/java/com/example/flight/controllers/FlightController.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+//    public Flight updateFlight(@RequestBody Flight flight, @PathVariable Long id) {
+//        return service.updateFlight(flight, id);
+//    }
+
+    @DeleteMapping("{id}")
+    public void deleteFlight(@PathVariable Long id) {
```
Comment (janvmusic) [inline src/main/java/com/example/flight/controllers/FlightController.java]: Users should not be able to remove flights. Why is this in the controller?

Think about what actions should be available in this controller

File: src/main/java/com/example/flight/exceptions/AirportCodeAlreadyExistAdvice.java [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@ControllerAdvice
+public class AirportCodeAlreadyExistAdvice {
+
+    @ResponseBody
+    @ExceptionHandler(AirportCodeAlreadyExistException.class)
+    @ResponseStatus(HttpStatus.I_AM_A_TEAPOT)
```
Comment (janvmusic) [inline src/main/java/com/example/flight/exceptions/AirportCodeAlreadyExistAdvice.java:14]: AHAHAHAHAHAHAAHA!!!!

Comment (janvmusic) [reply src/main/java/com/example/flight/exceptions/AirportCodeAlreadyExistAdvice.java:14]: Still use a correct value

File: src/main/java/com/example/flight/services/FlightService.java [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+    public FlightService(FlightRepository repository) {
+        this.repository = repository;
+    }
+
+    public List<Flight> getAllFlights() {
+        return repository.findAll();
+    }
+
+    public Flight createFlight(Flight flight) {
+        return repository.save(flight);
```
Comment (janvmusic) [inline src/main/java/com/example/flight/services/FlightService.java:27]: Remove, not required

File: src/main/java/com/example/flight/services/FlightService.java [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+    }
+
+    public Flight getFlightById(Long id) {
+        return repository.findById(id)
+                .orElseThrow(RuntimeException::new);
+    }
+
+//    public Flight updateFlight(Flight flight, Long id) {
```
Comment (janvmusic) [inline src/main/java/com/example/flight/services/FlightService.java]: Remove not required

File: src/main/java/com/example/flight/services/FlightService.java [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+//        return repository.findById(id)
+//                .map(flightFound -> {
+//                    flightFound.setAirport(flight.getAirport());
+//                    flightFound.setAvailable(flight.isAvailable());
+//                    return repository.save(flightFound);
+//                })
+//                .orElseThrow(RuntimeException::new);
+//    }
+
+    public void deleteFlight(Long id) {
```
Comment (janvmusic) [inline src/main/java/com/example/flight/services/FlightService.java:44]: remove not required

File: pom.xml [hunk 10]
```diff
@@ -29,7 +29,7 @@
 		<dependency>
 			<groupId>com.h2database</groupId>
```
Comment (janvmusic) [inline pom.xml:30]: Is this going to be replaced with postgres?

File: src/main/java/com/example/flight/LoadDatabase.java [hunk 11]
```diff
@@ -14,11 +16,20 @@
...
+            Airport airportB = new Airport("222B", "Ulau");
+            log.info("Preloading... " + airportRepository.save(airportA));
+            log.info("Preloading... " + airportRepository.save(airportB));
+            Flight flightA = new Flight(true, airportA);
+            Flight flightAA = new Flight(false, airportA);
+            Flight flightB = new Flight(true, airportB);
+            log.info("Preloading... " + flightRepository.save(flightA));
+            log.info("Preloading... " + flightRepository.save(flightAA));
+            log.info("Preloading... " + flightRepository.save(flightB));
+
```
Comment (janvmusic) [inline src/main/java/com/example/flight/LoadDatabase.java]: Either you give space between lines or remove this empty line

**Review by janvmusic — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/flight/controllers/FlightController.java (hunk 1 above)
Comment (JofielB) [reply src/main/java/com/example/flight/controllers/FlightController.java]: It was just a note to my self to remember where to continue the next day but I am gonna try to make more descriptive notes

File: src/main/java/com/example/flight/controllers/FlightController.java (hunk 4 above)
Comment (JofielB) [reply src/main/java/com/example/flight/controllers/FlightController.java]: This section was already changed

File: src/main/java/com/example/flight/LoadDatabase.java (hunk 11 above)
Comment (JofielB) [reply src/main/java/com/example/flight/LoadDatabase.java]: done

File: pom.xml (hunk 10 above)
Comment (JofielB) [reply pom.xml:30]: Yes, now the API is using Postgres

File: src/main/java/com/example/flight/controllers/FlightController.java (hunk 2 above)
Comment (JofielB) [reply src/main/java/com/example/flight/controllers/FlightController.java]: I think that have more sense to move this to the airport controller, so an airport can create its own flights

File: src/main/java/com/example/flight/controllers/FlightController.java (hunk 3 above)
Comment (JofielB) [reply src/main/java/com/example/flight/controllers/FlightController.java]: Sounds good to me, changed

File: src/main/java/com/example/flight/controllers/FlightController.java (hunk 5 above)
Comment (JofielB) [reply src/main/java/com/example/flight/controllers/FlightController.java]: I moved this method to the airport controller, it makes more sense for airports to have control over their flights

File: src/main/java/com/example/flight/services/FlightService.java (hunk 7 above)
Comment (JofielB) [reply src/main/java/com/example/flight/services/FlightService.java:27]: Why? How an airport will create a flight without this service?

File: src/main/java/com/example/flight/services/FlightService.java (hunk 8 above)
Comment (JofielB) [reply src/main/java/com/example/flight/services/FlightService.java]: This section is outdated

File: src/main/java/com/example/flight/services/FlightService.java (hunk 9 above)
Comment (JofielB) [reply src/main/java/com/example/flight/services/FlightService.java:44]: The airport cant delete a flight? what if a flight is created by mistake or they want to cancel it?

**Pull request merged**